compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -566,10 +566,132 @@ object Interactive {
+   *
+   * @param toName    The name that is set by the renaming.
+   * @param enclosing The tree that encloses the renaming import, if it exists.
+   * @param syms      The symbols to which we want to find renamed references.
+   * @param allTrees  All the trees in this source file, in case we can't find `enclosing`.
+   * @param source    The sourcefile that where to look for references.
+   * @return All the references to the symbol under the cursor that are using `toName`.
+   */
+  def findTreesMatchingRenaming(toName: Name,
+                                enclosing: Option[Tree],
+                                syms: List[Symbol],
+                                allTrees: List[Tree],
+                                source: SourceFile
+                               )(implicit ctx: Context): List[SourceNamedTree] = {
+
+    /**
+     * Remove the blocks that immediately enclose a renaming to `toName` in `inTree`.
+     *
+     * @param toName The target name of renamings.
+     * @param inTree The tree in which to remove the blocks that have such a renaming.
+     * @return A tree that has no children containing a renaming to `toName`.
+     */
+    def removeBlockWithRenaming(toName: Name, inTree: Tree): Tree = {
+      new TreeMap {
+        override def transform(tree: Tree)(implicit ctx: Context): Tree = tree match {
+          case pkg: PackageDef if immediatelyEnclosesRenaming(toName, pkg) =>
+            EmptyTree
+          case template: Template if immediatelyEnclosesRenaming(toName, template) =>
+            cpy.Template(template)(constr = DefDef(template.constr.symbol.asTerm), self = EmptyValDef, body = Nil)
+          case block @ Block(stats, expr) if immediatelyEnclosesRenaming(toName, block) =>
+            EmptyTree
+          case other =>
+            super.transform(other)
+        }
+      }.transform(inTree)
+    }
+
+    val trees = {
+      val trees = enclosing match {
+        case Some(pkg: PackageDef) =>
+          pkg.stats
+        case Some(template: Template) =>
+          template.body
+        case Some(block: Block) =>
+          block.expr :: block.stats
+        case _ =>
+          // No enclosing tree; we'll search in the whole file.
+          allTrees
+      }
+
+      // These trees may contain a new renaming of the same symbol to the same name, so we may
+      // have to cut some branches
+      val trimmedTrees = trees.map(removeBlockWithRenaming(toName, _))
+
+      // Some of these trees may not be `NameTrees`. Those that are not are wrapped in a
+      // synthetic val def, so that everything can go inside `SourceNamedTree`s.
+      trimmedTrees.map {
+        case tree: NameTree =>
+          SourceNamedTree(tree, source)
+        case tree =>
+          val valDef = tpd.SyntheticValDef(NameKinds.UniqueName.fresh(), tree)
+          SourceNamedTree(valDef, source)
+      }
+    }
+
+    val includes =
+      Include.references | Include.imports | Include.renamingImports
```

Comment:
Is there any situation where we have `Include.imports` but not `Include.renamingImports` ? Otherwise we should get rid of `renamingImports`

Reply:
Yes, this happens when we do `rename` on something that has not been renamed.